lib/ansible/modules/cloud/amazon/ec2.py

```diff
@@ -1438,6 +1441,24 @@ def startstop_instances(module, ec2, instance_ids, state, instance_tags):
                 inst.modify_attribute('disableApiTermination', termination_protection)
                 changed = True
 
+            # Check security groups and if we're using ec2-vpc; ec2-classic security groups may not be modified
+            if inst.vpc_id and group_name:
+                grp_details = ec2.get_all_security_groups(filters={'vpc_id': inst.vpc_id})
+                if isinstance(group_name, string_types):
+                    group_name = [group_name]
+                unmatched = set(group_name) - set(to_text(grp.name) for grp in grp_details)
+                if len(unmatched) > 0:
```

Comment:
note: tests for zero length and nonzero length containers can test the container directly:
``` python
# equivalent:
if unmatched: pass
if len(unmatched) == 0: pass

# also equivalent:
if not unmatched: pass
if len(unmatched) > 0: pass
```